modules/oe_subscriptions_digest/tests/src/Functional/AnonymousUserDigestTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_subscriptions_digest\Functional;
+
+use Drupal\decoupled_auth\Entity\DecoupledAuthUser;
+use Drupal\Tests\oe_subscriptions_anonymous\Trait\AssertMailTrait;
+use Drupal\user\UserInterface;
+
+/**
+ * Tests the user notifications frequency in subscriptions page.
+ */
+class AnonymousUserDigestTest extends UserDigestTest {
+
+  use AssertMailTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_subscriptions_anonymous',
+    'oe_subscriptions_digest',
+  ];
+
+  /**
+   * Tests the anonymous user digest.
+   */
+  public function testUserDigest(): void {
+    $user = DecoupledAuthUser::create([
+      'mail' => $this->randomMachineName() . '@example.com',
+      'name' => NULL,
+      'status' => 1,
+      'roles' => ['anonymous_subscriber'],
+    ]);
+    $user->save();
+    $fn_get_path = function (UserInterface $user) {
+      return $this->getAnonymousUserSubscriptionsPageUrl($user->getEmail());
+    };
+
+    $this->doTestDigestPreferences($user, $fn_get_path);
+    $this->doTestFlaggingDigest($user);
+  }
+
+  /**
+   * Retrieves the URL to access the subscriptions page for a user.
+   *
+   * This method will request the URL via the dedicated form.
+   *
+   * @param string $email
+   *   The e-mail of the user.
+   *
+   * @return string
+   *   The URL.
+   */
+  protected function getAnonymousUserSubscriptionsPageUrl(string $email): string {
```

Comment:
Is this duplicated? I didn't open the editor yet but I believe so, in that case we can have a trait.